src/qtcore/qml/elements/QtQuick/MouseArea.js

```diff
@@ -81,6 +83,16 @@ registerQmlType({
         self.containsMouse = false;
         self.exited();
     }
+    this.dom.onmouseover = function(e) {
+        self.containsMouse = true;
+        self.dom.style.cursor = cursorShapeToCSS();
+        self.entered();
+    }
+    this.dom.onmouseout = function(e) {
+        self.containsMouse = false;
+        self.dom.style.cursor = "auto";
+        self.exited();
+    }
```

Comment:
I've made a commit to update the behavior. I also found an issue with the `pressedButtons` property: in an event handler, it was refered as `this.pressedButtons`. But inside the event handlers, `this` isn't the QML component: that's why we have the `self` variable around, sometimes. I've also made a commit to fix that.